contributors/devel/development.md

```diff
@@ -211,41 +266,49 @@ code review tools can be very confusing.
 See [Faster Reviews](faster_reviews.md) for some thoughts on how to streamline
 the review process.
 
-### When to retain commits and when to squash
 
-Upon merge, all git commits should represent meaningful milestones or units of
-work.  Use commits to add clarity to the development and review process.
+#### Squash and Merge
 
-Before merging a PR, squash any "fix review feedback", "typo", and "rebased"
-sorts of commits. It is not imperative that every commit in a PR compile and
-pass tests independently, but it is worth striving for. For mass automated
-fixups (e.g. automated doc formatting), use one or more commits for the
-changes to tooling and a final commit to apply the fixup en masse. This makes
-reviews much easier.
+Upon merge (by either you or your reviewer), all commits left on the review
+branch should represent meaningful milestones or units of work.  Use commits to
+add clarity to the development and review process.
 
-## Testing
+Before merging a PR, squash any _fix review feedback_, _typo_, and _rebased_
+sorts of commits. 
 
-Three basic commands let you run unit, integration and/or e2e tests:
+It is not imperative that every commit in a PR compile and pass tests
+independently, but it is worth striving for.
 
-```sh
-cd kubernetes
-make test # Run every unit test
-make test WHAT=pkg/util/cache GOFLAGS=-v # Run tests of a package verbosely
-make test-integration # Run integration tests, requires etcd
-make test-e2e # Run e2e tests
-```
+For mass automated fixups (e.g. automated doc formatting), use one or more
+commits for the changes to tooling and a final commit to apply the fixup en
+masse. This makes reviews easier.
 
-See the [testing guide](testing.md) and [end-to-end tests](e2e-tests.md) for additional information and scenarios.
+#### Documentation extraction and release
 
-## Regenerating the CLI documentation
+At release time, documentation (e.g. the [kubectl user guide]) is extracted from
```

Comment:
This isn't really true anymore :(  It got complicated when we moved the documentation to a separate repo.  How much do you think we need to talk about this?